xattrvi

```diff
@@ -117,11 +118,15 @@ with open(tmpfile.name, 'w') as f:
 		f.write("# attributes in other namespaces:\n\n")
 		f.write("\n#".join([ "{0:10}   : {1}".format(key, value) for key, value in otherxattrs]))
 
+# code for editor quirks
+if re.findall('gvim',args.editor):
```

Comment:
Here's my reasoning behind the gvim regexp. When people generally have gvim set as their editor, it's usually called without a `--nofork` option. For gvim to work as an editor with xattrvi, the `nofork` option has to be set. So why not account for this use case inside of xattrvi? Otherwise, a user would try gvim and fail. After digging through the gvim man page, they would have to figure out that to use gvim with xattrvi, one has to use the `--nofork` option with gvim. If one accounts for this use case in the code, it makes it easier for the user. I guess it's a matter of philosophy.

re is part of the standard python library, so I'm not sure its a major dependency requirement.

Reply:
I disagree that we should start adding special cases for every single editor out there, because where do we draw the line then? So yes, this is inconvenient, but a `gvim`-problem and not to be handled in `xattrvi`. (`git` also does not handle this case, as I just checked, so `xattrvi` is definitely not the only software affected.)

I would, however, accept a section in the manpage mentioning such behavior for forking editors.

And yes, `re` is part of the standardlib, so not technically a dependency problem here. But not needing regex seems still desirable if circumventable. I would have preferred a simple `'gvim' in args.editor` also because it is much more readable as `re.findall(…)`. Not using modules if the same functionality can be achieved without it in a equivalently simple way smallens the code's footprint and makes it more mainainable imo.

Reply:
And I stand corrected, `args.editor.split()` is not a good idea, we need to `import shlex` and use `shlex.split(args.editor)`, because that handles quoting in quoted and things like this correctly.

So the correct lines would read

    import …,shlex
    …
    args.editor = shlex.split(args.editor)
    retval = subprocess.call(args.editor + [tmpfile.name])

See https://docs.python.org/3/library/shlex.html

Reply:
For some reason, I can't see my own previous comment, so I'll do it again: I'm not a hundred percent sure its best to say "its not our problem" here. gvim forks by design, and if xattrvi has to be useful from a graphical file manager, people might prefer gui based editors that tend to fork. But this is also pretty subjective, so I'm ok with moving the `--nofork` option to the desktop file and added it as an example to the README.